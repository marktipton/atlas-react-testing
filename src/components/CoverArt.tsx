import React from 'react'

type CoverArtProps = {
  cover: string;
  title: string;
};

function CoverArt({cover, title}: CoverArtProps) {
  return (
    <div>
      <img src={cover} alt={`${title} Cover`} className='rounded-lg w-full h-auto'/>
    </div>
  )
}

export default CoverArt;