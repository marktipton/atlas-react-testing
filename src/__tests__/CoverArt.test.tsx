import { render, screen } from "@testing-library/react";
import { expect, test } from "vitest";
import CoverArt from "../components/CoverArt";

test("CoverArt renders correctly", () => {
  const cover = "https://raw.githubusercontent.com/atlas-jswank/atlas-music-player-api/main/images/albumn6.webp";
  const title = "Album Title";

  const { container } = render(<CoverArt cover={cover} title={title} />);

  // Snapshot test
  expect(container).toMatchSnapshot();

  // Additional assertions
  const imgElement = screen.getByAltText(`${title} Cover`);
  expect(imgElement).toBeInTheDocument();
  expect(imgElement).toHaveAttribute('src', cover);
});