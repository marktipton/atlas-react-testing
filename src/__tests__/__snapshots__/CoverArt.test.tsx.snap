// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt renders correctly 1`] = `
<div>
  <div>
    <img
      alt="Album Title Cover"
      class="rounded-lg w-full h-auto"
      src="https://raw.githubusercontent.com/atlas-jswank/atlas-music-player-api/main/images/albumn6.webp"
    />
  </div>
</div>
`;
